### ISSUE TITLE
Assorted suggestions

### ISSUE THREAD

**Issue by markerikson** [taniarascia/takenote#4]
Hi!  Had a chance to glance through the code briefly, and wanted to offer up some suggestions:

- First, I'd strongly suggest that you try using [our Redux Starter Kit package](https://redux-starter-kit.js.org) once you feel like you understand the basics of Redux.  It should simplify a lot of the logic and code that you already have here.  For example, the entire `actions/index.ts` file can be removed completely, as can most of `constants/index.ts`, because RSK's `createSlice` function automatically generates action creators _and_ action types for you.
- Next, while Redux itself doesn't care about folder structures, our recommendation at this point is to use a "feature folder" approach, or even the "ducks" pattern for putting all logic for a given Redux feature into a single file.  See the [Redux FAQ entry on folder structures](https://redux.js.org/faq/code-structure#what-should-my-file-structure-look-like-how-should-i-group-my-action-creators-and-reducers-in-my-project-where-should-my-selectors-go) for comparisons, and the [RSK usage guide on `createSlice`](https://redux-starter-kit.js.org/usage/usage-guide#exporting-and-using-slices) for some discussion of using RSK to write "ducks" files.
- Finally, I see that you're using sagas here.  Sagas are a great power tool, but the majority of Redux apps don't need them.  In your case, it looks like all you're doing is fairly simple data fetching.  This can easily be done with Redux "thunks" and async functions.  For example, this saga:

```ts
function* fetchNotes() {
  try {
    const notes = yield requestNotes()

    yield put(loadNotesSuccess(notes))
  } catch (error) {
    yield put(loadNotesError(error.message))
  }
}

export function* allSagas() {
  yield all([
    takeLatest(Actions.LOAD_NOTES, fetchNotes),
  ])
}
```

becomes this thunk:

```ts
const fetchNotes = () => async dispatch => {
    try {
        const notes = await dispatch(requestNotes());
        dispatch(loadNotesSuccess(notes);
    } catch(error) {
        dispatch(loadNotesError(error.messsage);
    }
}
```

So, sagas are definitely not _necessary_ here, and thunks should be simpler for this use case.

If you use RSK's `configureStore` function, it adds the thunk middleware to the store by default.

See the [Redux FAQ entry on choosing an async middleware](https://redux.js.org/faq/actions#what-async-middleware-should-i-use-how-do-you-decide-between-thunks-sagas-observables-or-something-else) for more info, and the [RSK "Advanced" tutorial"](https://redux-starter-kit.js.org/tutorials/advanced-tutorial) shows how to do data fetching with thunks and use it with TS.

One other suggestion: while `connect` is our main API for interacting with the store from components, we also now have [a React-Redux hooks API as well](https://react-redux.js.org/api/hooks), which may be simpler to work with in terms of lines of code and TS usage.  You might want to try those out.

Hopefully these suggestions are useful.  Please let me know if you've got any questions I can help with!

**Comment by taniarascia** [comment]
Hi Mark! Thank you so much for your comment and suggestions. I will definitely take a look at the naming conventions and play around with it. I’m kind of learning as I go, so I’m probably introducing some tedious methods of solving problems.

You’re right, Thunk does look simpler than Sagas. I looked at both and went with Sagas because they seem to be pretty popular and beloved, but it does seem like overkill for my simple use case. Thunk is a lot more familiar to the way I’m used to doing things.

Again, thanks a bunch and I’ll consider your suggestions. It’s an honor to have to critique my humble little project. 🙂

**Comment by MartinRosenberg** [comment]
I'm gonna take a crack at doing this with RSK.

**Comment by MartinRosenberg** [comment]
Thanks again for these suggestions, @markerikson! 

@taniarascia has decided to leave `redux-saga` where it is; the code is still pretty simple (and not really any more complicated than the `redux-thunk` code you shared), so we may as well keep the power tool.

I've started the conversion to RSK with #35 and it's already a good deal cleaner. Some more reorganization is required for the feature-folder/ducks approach, but I'm definitely gonna try to do that too. I also might write a blog post about the switch, and help/encourage others to do the same.

**Comment by markerikson** [comment]
Sure!  Nothing wrong with wanting to keep the sagas if you're comfortable with them.  Just wanted to point out that thunks are the recommended default option, and that the sagas certainly aren't _necessary_ in this case, as an FYI.

Please let me know when you've got the conversion done and that blog post up.  I'd be happy to provide feedback and help publicize it.

Also, note that I just published RSK v0.8 last night, which has a small breaking change (`slice?` -> `name`), and I'm hoping to get 1.0 out within the next week or so.
